Assets/Scripts/Player Scripts/PlayerController.cs

```diff
@@ -183,8 +201,33 @@ private bool IsOnGround()
         Vector2 direction = Vector2.down;
         float distance = groundCheckDistance;
 
```

Comment:
Using an epsilon value (groundCheckPreipheralDistance) to approximate the width of your collider is bad. Here's how you should go about checking for ground collision.

Using a reference to your player's box collider component, use the Collider2D.Bounds function to get a Bound type that contains the world space information of your collider

Cast 2 rays, one from the bottom left corner, (bounds.min) and one from the bottom right corner (new Vector2(bounds.max.x, bounds.min.y))

These rays will originate from the bottom corners of your player's box collider no matter what size it is. You can scale it at runtime and the ray will always originate at the very corner.

Finally, simply perform an if check if either of the rays is "hitting" the ground. If at least one is, the player is grounded. If both are not hitting the ground, they are falling.